src/MagnumPlugins/TinyGltfImporter/TinyGltfImporter.cpp

```diff
@@ -0,0 +1,484 @@
+    } else if(node.extLightsValues.find("light") != node.extLightsValues.end()) {
+        UnsignedInt lightId = node.extLightsValues.find("light")->second.number_array[0];
+
+        return std::unique_ptr<ObjectData3D>{new ObjectData3D{children, transformation, ObjectInstanceType3D::Light, lightId, &node}};
+    }
+
+    return std::unique_ptr<ObjectData3D>{new ObjectData3D{children, transformation, &node}};
+}
+
+UnsignedInt TinyGltfImporter::doMesh3DCount() const {
+    return _d->model.meshes.size();
+}
+
+Containers::Optional<MeshData3D> TinyGltfImporter::doMesh3D(const UnsignedInt id) {
+    const tinygltf::Mesh& mesh = _d->model.meshes[id];
+
+    /* primitive */
+    MeshPrimitive meshPrimitive = MeshPrimitive::Triangles;
+
+    if (mesh.primitives.size() > 1) {
+        Error() << "Trade::TinyGltfImporter::mesh3D(): currently only one primitive per mesh is supported";
+        return Containers::NullOpt;
+    }
+
+    const tinygltf::Primitive& primitive = mesh.primitives[0];
+
+    int mode = primitive.mode;
+    if (mode == TINYGLTF_MODE_POINTS) {
+        meshPrimitive = MeshPrimitive::Points;
+    } else if (mode == TINYGLTF_MODE_LINE) {
+        meshPrimitive = MeshPrimitive::Lines;
+    }
+
+    /* Vertices */
+    std::vector<std::vector<Vector3>> positions;
+    std::vector<std::vector<Vector2>> textureCoordinates;
+    std::vector<std::vector<Vector3>> normals;
+
+    const tinygltf::Accessor& posAccessor = _d->model.accessors[primitive.attributes.find("POSITION")->second];
+    const tinygltf::BufferView& posBufferview = _d->model.bufferViews[posAccessor.bufferView];
+    const tinygltf::Buffer& posBuffer = _d->model.buffers[posBufferview.buffer];
+
+    if (posAccessor.type != TINYGLTF_TYPE_VEC3) {
+        Error() << "Trade::TinyGltfImporter::mesh3D(): expected type of vertex positions is VEC3";
+        return Containers::NullOpt;
+    }
+
+    std::vector<Vector3> posElement;
+    size_t numPositions = posBufferview.byteLength / sizeof(Vector3);
+    posElement.resize(numPositions);
+    std::memcpy(posElement.data(), posBuffer.data.data() + posBufferview.byteOffset, posBufferview.byteLength);
+    positions.push_back(std::move(posElement));
+
+    const tinygltf::Accessor& normalAccessor = _d->model.accessors[primitive.attributes.find("NORMAL")->second];
+    const tinygltf::BufferView& normalBufferview = _d->model.bufferViews[normalAccessor.bufferView];
+    const tinygltf::Buffer& normalBuffer = _d->model.buffers[normalBufferview.buffer];
+
+    if (normalAccessor.type != TINYGLTF_TYPE_VEC3) {
+        Error() << "Trade::TinyGltfImporter::mesh3D(): expected type of normal is VEC3";
+        return Containers::NullOpt;
+    }
+
+    std::vector<Vector3> normalElement;
+    size_t numNormals = normalBufferview.byteLength / sizeof(Vector3);
+    normalElement.resize(numNormals);
+    std::memcpy(normalElement.data(), normalBuffer.data.data() + normalBufferview.byteOffset, normalBufferview.byteLength);
```

Comment:
Why sometimes `std::memcpy()` and sometimes `std::copy()`? I would prefer `std::copy()` in all cases. Also, there's `std::copy_n()` if you'd want to avoid calling `reinterpret_cast` on both arguments.